cmd/traffic/cmd/manager/internal/state/intercept.go

```diff
@@ -355,10 +341,68 @@ func waitForConfigMapUpdate(ctx context.Context, cmAPI typed.ConfigMapInterface,
 	}
 }
 
-func (s *State) waitForAgent(ctx context.Context, name, namespace string) error {
+func watchFailedInjectionEvents(ctx context.Context, name, namespace string) (<-chan *events.Event, error) {
+	// A timestamp with second granularity is needed here, because that's what the event creation time uses.
+	// Finer granularity will result in relevant events seemingly being created before this timestamp because
+	// they have the fraction of seconds trimmed off (which is odd, given that the type used is a MicroTime).
+	start := time.Unix(time.Now().Unix(), 0)
+
+	ei := k8sapi.GetK8sInterface(ctx).EventsV1().Events(namespace)
+	w, err := ei.Watch(ctx, meta.ListOptions{
+		FieldSelector: fields.OneTermNotEqualSelector("type", "Normal").String()})
```

Comment:
could this be further filtered down to only get events related to the agent?

Reply:
The filtering mechanism is very limited. It only allows for exact matches and only AND conditions can be used. That why I decided to just match abnormal events here and then do more advanced filtering further down in this method. It shouldn't matter much unless the cluster is in a very bad shape and sends a very large number of abnormal events.